File: data/transcoder_evaluation_gfg/go/FIND_NUMBER_PERFECT_SQUARES_TWO_GIVEN_NUMBERS.go
package main

import (
		"fmt"
	"os"
	"unsafe"
)

func min(x int, y int) int {
	if x < y {
		return x
	}
	return y
}
func max(x int, y int) int {
	if x > y {
		return x
	}
	return y
}
func cmpfunc(a unsafe.Pointer, b unsafe.Pointer) int {
	return *(*int)(a) - *(*int)(b)
}


func f_gold(a int, b int) int {
	var cnt int = 0
	for i := int(a); i <= b; i++ {
		for j := int(1); j*j <= i; j++ {
			if j*j == i {
				cnt++
			}
		}
	}
	return cnt
}
//TOFILL
func main() {
	var (
		n_success int     = 0
		param0    []int = []int{48, 3, 20, 98, 96, 40, 9, 57, 28, 98}
		param1    []int = []int{42, 82, 72, 98, 90, 82, 15, 77, 80, 75}
	)
	for i := int(0); i < len(param0[:]); i++ {
		if f_filled(param0[i], param1[i]) == f_gold(param0[i], param1[i]) {
			n_success += 1
		}
	}
	fmt.Print("#Results:", " ", n_success, ", ", len(param0[:]))
	os.Exit(0)
}

File: data/transcoder_evaluation_gfg/rust/FIND_NUMBER_PERFECT_SQUARES_TWO_GIVEN_NUMBERS.rs
#![feature(main)]
// Copyright (c) 2019-present, Facebook, Inc.
// All rights reserved.
//
// This source code is licensed under the license found in the
// LICENSE file in the root directory of this source tree.
//

#[no_mangle]
pub unsafe fn abs(mut x: i32)
 -> i32 {
    return x.abs();
}

#[no_mangle]
pub unsafe fn min(mut x: i32, mut y: i32)
 -> i32 {
    return if x < y { x } else { y };
}
#[no_mangle]
pub unsafe fn max(mut x: i32, mut y: i32)
 -> i32 {
    return if x > y { x } else { y };
}

pub unsafe fn f_gold(mut a: i32, mut b: i32)
 -> i32 {
    let mut cnt: i32 = 0 as i32;
    let mut i: i32 = a;
    while i <= b {
        let mut j: i32 = 1 as i32;
        while j * j <= i { if j * j == i { cnt += 1 } j += 1 }
        i += 1
    }
    return cnt;
}
//TOFILL
unsafe fn main_0() -> i32 {
    let mut n_success: i32 = 0 as i32;
    let mut param0: [i32; 10] =
        [48 as i32, 3 as i32, 20 as i32,
         98 as i32, 96 as i32, 40 as i32,
         9 as i32, 57 as i32, 28 as i32,
         98 as i32];
    let mut param1: [i32; 10] =
        [42 as i32, 82 as i32, 72 as i32,
         98 as i32, 90 as i32, 82 as i32,
         15 as i32, 77 as i32, 80 as i32,
         75 as i32];
    let mut i: i32 = 0 as i32;
    while i < param0.len() as i32 {
        if f_filled(param0[i as usize], param1[i as usize]) ==
               f_gold(param0[i as usize], param1[i as usize]) {
            n_success += 1 as i32
        }
        i += 1
    }
    println!("{} {} {} {} {}", "#Results:", " ", n_success, ", ", param0.len() as i32);
    return 0 as i32;
}
#[main]
pub fn main() { unsafe { ::std::process::exit(main_0() as i32) } }
